scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2109,37 @@ def visit_call(self, node):
                     break
 
 
+class MetaclassChecker(checkers.BaseChecker):
+    """Custom pylint checker prohibiting use of "__metaclass__" and
+    enforcing use of "python_utils.with_metaclass()" instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'replace-disallowed-metaclass-usage'
+    priority = -1
+    msgs = {
+        'C0034': (
+            'Please use python_utils.with_metaclass() instead of __metaclass__',
+            'metaclass',
+            'Enforce usage of python_utils.with_metaclass()'
+            'instead of __metaclass__'
+        )
+    }
+
+    def visit_classdef(self, node):
+        """Visit each class definition in a module and check if there is a
+        __metaclass__ present.
+
+        Args:
+            node: astroid.nodes.ClassDef. Node for a class definition
+                in the AST.
+        """
+        if node.declared_metaclass() is None:
+            return
+        self.add_message('replace-disallowed-metaclass-usage', node=node)
```

Comment:
The message ID on top is `metaclass` and we are using `'replace-disallowed-metaclass-usage'` then how is it working? Is this expected?

Reply:
Changed the LOGIC